exec/java-exec/src/main/java/org/apache/drill/exec/expr/ComparisonPredicate.java

```diff
@@ -88,11 +88,11 @@ private ComparisonPredicate(LogicalExpression left,
   @Override
   @SuppressWarnings("unchecked")
   public RowsMatch matches(StatisticsProvider<C> evaluator) {
-    ColumnStatistics leftStat = left.accept(evaluator, null);
+    ColumnStatistics<BigInteger> leftStat = (ColumnStatistics<BigInteger>) left.accept(evaluator, null);
```

Comment:
Actually, the value here may be not only with `BigInteger` type. Please use `C` for generic and do the cast after the check for `VARDECIMAL` type.